packages/core/src/components/icon/icon.tsx

```diff
@@ -7,60 +7,63 @@
-     * Constants are exposed for each of these values on the component itself:
-     * `Icon.SIZE_(STANDARD|LARGE|INHERIT)`,
-     * @default 16
+     * Size of the icon, in pixels.
+     * Blueprint contains 16px and 20px SVG icon images,
+     * and chooses the appropriate resolution based on this prop.
      */
-    iconSize?: 16 | 20 | "inherit";
+    iconSize?: number;
+
+    /** CSS style properties. */
+    style?: React.CSSProperties;
 }
 
-export class Icon extends React.PureComponent<IIconProps & React.HTMLAttributes<HTMLSpanElement>, never> {
+export class Icon extends React.PureComponent<IIconProps & React.SVGAttributes<SVGElement>> {
     public static displayName = "Blueprint.Icon";
 
-    public static readonly SIZE_STANDARD = 16 as 16;
-    public static readonly SIZE_LARGE = 20 as 20;
-    public static readonly SIZE_INHERIT = "inherit" as "inherit";
+    public static readonly SIZE_STANDARD = 16;
+    public static readonly SIZE_LARGE = 20;
 
     public render() {
-        if (this.props.iconName == null) {
+        const { className, iconName, iconSize = Icon.SIZE_STANDARD, intent, ...svgProps } = this.props;
+        if (iconName == null) {
             return null;
         }
-        const { className, iconName, intent, iconSize = Icon.SIZE_STANDARD, ...restProps } = this.props;
-
-        const classes = classNames(
-            getSizeClass(iconSize),
-            Classes.iconClass(iconName),
-            Classes.intentClass(intent),
-            className,
+        // choose which pixel grid is most appropriate for given icon size
+        const pixelGridSize = iconSize <= Icon.SIZE_STANDARD ? Icon.SIZE_STANDARD : Icon.SIZE_LARGE;
+        const classes = classNames(Classes.ICON, Classes.iconClass(iconName), Classes.intentClass(intent), className);
+        const viewBox = `0 0 ${pixelGridSize} ${pixelGridSize}`;
+        return (
+            <svg {...svgProps} className={classes} width={iconSize} height={iconSize} viewBox={viewBox}>
+                <title>{iconName}</title>
+                {this.renderSvgPaths(pixelGridSize)}
+            </svg>
         );
-        return <span className={classes} {...restProps} />;
     }
-}
 
-// NOTE: not using a type alias here so the full union will appear in the interface docs
-function getSizeClass(size: 16 | 20 | "inherit") {
-    switch (size) {
-        case Icon.SIZE_STANDARD:
-            return Classes.ICON_STANDARD;
-        case Icon.SIZE_LARGE:
-            return Classes.ICON_LARGE;
-        default:
-            return Classes.ICON;
+    private renderSvgPaths(pathsSize: number) {
+        const svgPaths = pathsSize === Icon.SIZE_STANDARD ? IconSvgPaths16 : IconSvgPaths20;
+        const paths = svgPaths[this.props.iconName.replace("pt-icon-", "") as IconName];
+        if (paths == null) {
+            return null;
+        }
+        return paths.map((d, i) => <path key={i} d={d} fillRule="evenodd" />);
```

Comment:
please document the usage of `fillRule`

Reply:
@pkwi do you know why this property is here? it came from the original SVG files. is it necessary?

Reply:
yes, this is necessary. `fillRule` is an attribute used in SVG that determines what paths are inside the shape and calculates how the shape is filled. Even-odd counts the amount of paths from the outside, if the path is an odd number, it gets subtracted - it will preserve the holes in overlapping shapes.

Reply:
however, there is a difference between SVG icons exported in Illustrator and Sketch. Illustrator places `fillRule` within the `path` where Sketch is placing it within the `group` container. I dont believe there will be too much harm if we remove it.

Reply:
this is the only place it's defined for the SVG icons. if the `fillRule` property is necessary, then it has to stay right here. 

https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/fill-rule